rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/undo/UndoLogManager.java

```diff
@@ -42,19 +41,40 @@
  * The type Undo log manager.
  */
 public final class UndoLogManager {
+    private enum State {
+        /**
+         * This state can be properly rolled back by fescar
+         */
+        Normal(0),
+        /**
+         * This state prevents the branch transaction from inserting undo_log after the global transaction is rolled
+         * back.
+         */
+        GlobalFinished(1);
+
+        private int value;
+
+        State(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+    }
 
     private static final Logger LOGGER = LoggerFactory.getLogger(UndoLogManager.class);
 
     private static String UNDO_LOG_TABLE_NAME = "undo_log";
 
     private static String INSERT_UNDO_LOG_SQL = "INSERT INTO " + UNDO_LOG_TABLE_NAME + "\n" +
         "\t(branch_id, xid, rollback_info, log_status, log_created, log_modified)\n" +
-        "VALUES (?, ?, ?, 0, now(), now())";
+        "VALUES (?, ?, ?, ?, now(), now())";
     private static String DELETE_UNDO_LOG_SQL = "DELETE FROM " + UNDO_LOG_TABLE_NAME + "\n" +
         "\tWHERE branch_id = ? AND xid = ?";
 
     private static String SELECT_UNDO_LOG_SQL = "SELECT * FROM " + UNDO_LOG_TABLE_NAME
```

Comment:
final?

Reply:
yes, any problems?

Reply:
> yes, any problems?

It is recommended that you use final keyword for a constant .

Reply:
It is recommended that you use final keyword for a constant .

Reply:
This has nothing to do with this issue, and it is related to checkstyle. Does the checkstyle related problem need to be handled separately?